Update readme (#1)
DevExpress-Examples/asp-net-web-forms-gridview-change-a-cell-value-based-on-another-cell-value-in-batch-edit-mode: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T558750)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# GridView for Web Forms - How to change a cell value based on another cell value in batch edit mode

> The example demonstrates how to change a column value (Price) when another column (Percentage) is changed and vice versa.

Reviewer: ```suggestion
This example demonstrates how to change a column value (Price) when another column (Percentage) is changed and vice versa.
```

![Grid View - Changed Values](grid-batch-linked-values.png)

> [ASPxClientGridView](https://docs.devexpress.com/AspNet/js-ASPxClientGridView) class implements the [batchEditApi](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.batchEditApi) property that exposes API to work with grid in batch mode on the client side.

Reviewer: ```suggestion
The [ASPxClientGridView](https://docs.devexpress.com/AspNet/js-ASPxClientGridView) class implements the [batchEditApi](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.batchEditApi) property that exposes an API to work with the grid in batch mode on the client side.
```

> When value editing is over, the [BatchEditEndEditing](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event is raised. In the event handler, you can get a cell value in the following ways:

Reviewer: `When value editing is over`

I'd like to specify _who_ edits the value if possible. Is it a user or developer?

Author: ```suggestion
When a user finishes value editing, the [BatchEditEndEditing](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event is raised. In the event handler, you can get a cell value in the following ways:
```

Reviewer: How about the following?

`After a user edits the value, the [BatchEditEndEditing](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event is raised. In the event handler, you can get a cell value in the following ways:`

Author: ```suggestion
After a user edits the value, the [BatchEditEndEditing](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event is raised. In the event handler, you can get a cell value in the following ways:
```

> - [batchEditApi.GetCellValue](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewBatchEditApi.GetCellValue(visibleIndex-columnFieldNameOrId)) method returns a value of a cell that is not in **editing mode**.

Reviewer: 1. Please add an article ("The") before "batchEditApi.GetCellValue".

2. I noticed that the phrase "a value of a cell" is singular. Can _only_ one cell be a cell that is not in edit mode? Please clarify for my understanding.

Author: 2. I mean "a value of any cell", but now I see that it is better to say "the value of the specified cell"

```suggestion
- The [batchEditApi.GetCellValue](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewBatchEditApi.GetCellValue(visibleIndex-columnFieldNameOrId)) method returns the value of the specified cell that is not in **editing mode**.
```

Reviewer: I like your suggestion, but can we change "editing mode" to "edit mode"?

Reviewer: ```suggestion
- The [batchEditApi.GetCellValue](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewBatchEditApi.GetCellValue(visibleIndex-columnFieldNameOrId)) method returns the value of the specified cell that is not in **edit mode**.
```

> - [rowValues](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewBatchEditEndEditingEventArgs.rowValues) property returns a new entered value of a cell that is in **editing mode**.

Reviewer: 1. Please add an article ("The") before "rowValues".

2. I think "a new entered value" should be "the newly entered value".
However, is it necessary to say "newly entered"? For example, can we omit this phrase and say:

`- The [rowValues](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewBatchEditEndEditingEventArgs.rowValues) property returns the value of a cell that is in **edit mode**.`

Author: I prefer to remain "the newly entered value" (or just "the new value"), because the GetCellValue method can also return the value of a cell that is in **edit mode**, but it returns the old value.
```suggestion
- The [rowValues](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewBatchEditEndEditingEventArgs.rowValues) property returns the new value of the specified cell that is in **editing mode**.
```

Reviewer: Thank you for the suggestion. Can we change "editing mode" to "edit mode"?

```suggestion
- The [rowValues](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewBatchEditEndEditingEventArgs.rowValues) property returns the new value of the specified cell that is in **edit mode**.
```
